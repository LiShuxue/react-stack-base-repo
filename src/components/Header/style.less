// use nesting style , or it will be global class
.header {
  display: flex;
  align-items: center;
  padding: 0 20px;
  width: 100%;
  background-color: var(--theme-color-header-bg);
  border-bottom: 5px solid var(--theme-color-header-border);

  .header-nav {
    margin-left: 30px;
    .link {
      display: inline-block;
      height: 48px;
      line-height: 48px;
      padding: 0 10px;
      margin-right: 20px;
      color: var(--theme-color-header-font);
      font-size: 13px;
      text-decoration: none;
      &:hover {
        color: var(--theme-color-header-font-hover);
      }
      &.active {
        position: relative;
        background: var(--theme-color-header-active-bg);
      }
    }
  }

  a.active::before {
    content: '';
    display: inline-block;
    height: 51px;
    width: 18px;
    position: absolute;
    left: -12px;
    background: var(--theme-color-header-active-bg);
    transform: rotate(15deg);
  }
  a.active::after {
    content: '';
    display: inline-block;
    height: 51px;
    width: 18px;
    position: absolute;
    right: -12px;
    background: var(--theme-color-header-active-bg);
    transform: translateY(-2px) rotate(15deg);
  }

  .header-icon {
    height: 36px;
  }
}
