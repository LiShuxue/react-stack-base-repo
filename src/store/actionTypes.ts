export const ACTION_TYPE_A: string = 'ACTION_TYPE_A';
export const ACTION_TYPE_B: string = 'ACTION_TYPE_B';
