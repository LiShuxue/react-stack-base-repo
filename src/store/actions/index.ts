import { actionCreatorA } from './actionCreatorA';
import { asyncActionCreatorB } from './actionCreatorB';

export default {
  actionCreatorA,
  asyncActionCreatorB
};
