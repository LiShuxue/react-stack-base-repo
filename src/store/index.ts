import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducer from './reducers';

// preload state object key must be same as reducer
const preloadState = {
  stateA: 'this is preload A',
  stateB: 'this is preload B'
};

// we not have to pass the preloadState, then it will use the default value in reducer as the state
const store = createStore(reducer, preloadState, applyMiddleware(thunk));

export default store;