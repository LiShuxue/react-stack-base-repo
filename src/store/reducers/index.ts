import { combineReducers } from 'redux';

import { handleActionA } from './moduleAReducer';
import { handleActionB } from './moduleBReducer';

// preload state object key must be same as reducer
const reducer = combineReducers({
  stateA: handleActionA,
  stateB: handleActionB
})

export default reducer;