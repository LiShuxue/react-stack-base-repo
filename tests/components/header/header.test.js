import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from 'components/header';
import { Provider } from 'react-redux';
import store from 'src/store';

test('should render Header component correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('should have "page1", "page2", "page3" link', () => {
  const { getByText } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );

  expect.assertions(6);

  const page1 = getByText('page1');
  expect(page1).toBeInTheDocument();
  expect(page1).toHaveClass('link');

  const page2 = getByText('page2');
  expect(page2).toBeInTheDocument();
  expect(page2).toHaveClass('link');

  const page3 = getByText('page3');
  expect(page3).toBeInTheDocument();
  expect(page3).toHaveClass('link');
});

test('if link be choose, should have active class', () => {
  const { getByText } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );

  expect.assertions(3);

  const page1 = getByText('page1');
  fireEvent.click(page1);
  expect(page1).toHaveClass('active');

  const page2 = getByText('page2');
  fireEvent.click(page2);
  expect(page2).toHaveClass('active');

  const page3 = getByText('page3');
  fireEvent.click(page3);
  expect(page3).toHaveClass('active');
});
