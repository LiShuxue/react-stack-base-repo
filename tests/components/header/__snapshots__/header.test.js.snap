// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<div
  className="header"
>
  <img
    alt="Ratan One"
    className="header-icon"
    src="logo.png"
  />
  <div
    className="header-nav"
  >
    <a
      aria-current={null}
      className="link"
      href="/query"
      onClick={[Function]}
    >
      Trade Query
    </a>
    <a
      aria-current={null}
      className="link"
      href="/exceptions"
      onClick={[Function]}
    >
      Exceptions
    </a>
    <a
      aria-current={null}
      className="link"
      href="/rules"
      onClick={[Function]}
    >
      Rules
    </a>
  </div>
</div>
`;
